src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -263,85 +262,101 @@ public IEnumerable<PackageItemViewModel> GetCurrent()
+                        Vulnerabilities = metadataContextInfo.Vulnerabilities,
+                        Sources = _packageSources,
+                        PackagePath = metadataContextInfo.PackagePath,
+                        PackageFileService = _packageFileService,
+                        IncludePrerelease = _includePrerelease,
+                        PackageLevel = packageLevel,
+                        TransitiveToolTipMessage = transitiveToolTipMessage,
+                        InstalledVersions = new[] { metadataContextInfo.Identity.Version },
+                    };
+
+                    if (listItem.PackageLevel == PackageLevel.TopLevel)
+                    {
+                        listItem.UpdatePackageStatus(_installedPackages);
+                    }
+                    else
+                    {
+                        listItem.UpdateTransitivePackageStatus(metadataContextInfo.Identity.Version);
+                    }
 
-                var listItem = new PackageItemViewModel(_searchService, _packageVulnerabilityService)
-                {
-                    Id = metadataContextInfo.Identity.Id,
-                    Version = metadataContextInfo.Identity.Version,
-                    IconUrl = metadataContextInfo.IconUrl,
-                    Owner = metadataContextInfo.Owners,
-                    KnownOwnerViewModels = knownOwnerViewModels,
-                    Author = metadataContextInfo.Authors,
-                    DownloadCount = metadataContextInfo.DownloadCount,
-                    Summary = metadataContextInfo.Summary,
-                    AllowedVersions = allowedVersions,
-                    VersionOverride = versionOverride,
-                    PrefixReserved = metadataContextInfo.PrefixReserved && !IsMultiSource,
-                    Recommended = metadataContextInfo.IsRecommended,
-                    RecommenderVersion = metadataContextInfo.RecommenderVersion,
-                    Vulnerabilities = metadataContextInfo.Vulnerabilities,
-                    Sources = _packageSources,
-                    PackagePath = metadataContextInfo.PackagePath,
-                    PackageFileService = _packageFileService,
-                    IncludePrerelease = _includePrerelease,
-                    PackageLevel = packageLevel,
-                    TransitiveToolTipMessage = transitiveToolTipMessage,
-                };
-
-                if (packageLevel == PackageLevel.TopLevel)
-                {
-                    listItem.UpdatePackageStatus(_installedPackages);
-                }
-                else
-                {
-                    listItem.UpdateTransitivePackageStatus(metadataContextInfo.Identity.Version);
+                    listItemViewModels[packageId] = listItem;
                 }
+            }
 
-                listItemViewModels.Add(listItem);
+            foreach (var item in listItemViewModels.Values)
+            {
+                item.UpdateInstalledPackagesVulnerabilities();
```

Comment:
UpdatePackageStatus and UpdateTransitivePackageStatus will both fetch vulnerability details for the package as it gets created. This new method seems to delegate to ReloadPackageVulnerabilitiesAsync which appears to fetch vulnerability data for all installed packages. Do we need to do this in both places? Can we fetch vulnerability data in only one place?